apis/v1alpha1/route_types.go

```diff
@@ -19,16 +19,39 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// GatewayObjectReference identifies a Gateway object.
-type GatewayObjectReference struct {
-	// Namespace is the namespace of the referent.
+// GatewayAllowType specifies which Gateways should be allowed to use a Route.
+type GatewayAllowType string
+
+const (
+	// GatewayAllowAll indicates that all Gateways will be able to use this
+	// route.
+	GatewayAllowAll GatewayAllowType = "All"
+	// GatewayAllowFromList indicates that only Gateways that have been
+	// specified in GatewayRefs will be able to use this route.
+	GatewayAllowFromList GatewayAllowType = "FromList"
+	// GatewayAllowSameNamespace indicates that only Gateways within the same
+	// namespace will be able to use this route.
+	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
+)
+
+// RouteGateways defines which Gateways will be able to use a route. If this
+// field results in preventing the selection of a Route by a Gateway, an
+// "Admitted" condition with a status of false must be set for the Gateway on
+// that Route.
+type RouteGateways struct {
+	// Allow indicates which Gateways will be allowed to use this route.
+	// Possible values are:
+	// * All: Gateways in any namespace can use this route.
+	// * FromList: Only Gateways specified in GatewayRefs may use this route.
+	// * SameNamespace: Only Gateways in the same namespace may use this route.
+	// +kubebuilder:validation:Enum=All;List;SameNamespace
```

Comment:
s/List/FromList/

Reply:
Updated, thanks!

Reply:
Re-opening this thread, the diff still shows `List` for me.

Reply:
🤦 Fix should actually be pushed now...